### PR TITLE
Add support for pooler load balancer

### LINKED ISSUE #940: Add support for pooler LoadBalancer
There are currently options to enable load balancers for master and replica nodes, but that would be cool to support balancing for pooler service too.

### PR DESCRIPTION
**Pull request by Prototik**
Closes #940

### CODE REVIEW

File: charts/postgres-operator/values-crd.yaml [hunk 1]
```diff
@@ -163,6 +163,8 @@ configLoadBalancer:
   enable_master_load_balancer: false
   # toggles service type load balancer pointing to the replica pod of the cluster
   enable_replica_load_balancer: false
+  # toggles service type load balancer pointing to the pooler pod of the cluster
+  enable_pooler_load_balancer: false
```
Comment (Jan-M) [inline charts/postgres-operator/values-crd.yaml:167]: We did not consider the master/replica in the pooler yet, but probably good idea to make this `enable_master_pooler_load_balancer`

Comment (Prototik) [reply charts/postgres-operator/values-crd.yaml:167]: Wait, pooler works only with master nodes? Shouldn't it delegate write requests to master and read request to all nodes?

Comment (Jan-M) [reply charts/postgres-operator/values-crd.yaml:167]: This is not possible. This needs to be a conscious decision in your application. Supporting the pooler in front of the replicas will follow.

PGBouncer cannot decide this. Imagine a transaction spanning multiple commands and starting only with reads and ending with a write, where would you point it to.

Comment (Prototik) [reply charts/postgres-operator/values-crd.yaml]: Oh, that makes sense. Renamed.

**Comment by sdudoladov** [comment]
hello @Prototik 

1. Please add example to [the complete Postgres manifest](https://github.com/zalando/postgres-operator/blob/master/manifests/complete-postgres-manifest.yaml#L114)
2. This setting should be grouped with other [connection pooler settings](https://github.com/zalando/postgres-operator/blob/f32c615a531426804d031b16967d69a4ec364c16/pkg/apis/acid.zalan.do/v1/postgresql_type.go#L193) both in he example and the code

**Comment by Jan-M** [comment]
Looks good to me if we get the tiny requested changes in,

File: charts/postgres-operator/values-crd.yaml (hunk 1 above)
Comment (ntcong) [reply charts/postgres-operator/values-crd.yaml]: pgpool do support routing read-only request to slave.

**Comment by michaelmcmillan** [comment]
Did this never get merged in?

**Pull request closed without merging**